src/components/GameView/GameMenu.vue

```diff
@@ -132,6 +143,11 @@ export default {
       this.loading = false;
       this.showStalemateDialog = false;
     },
+    async stopSpectate() {
+      this.loading = true;
+      await this.$store.dispatch('requestSpectateLeave');
+      this.$router.push('/');
```

Comment:
Just to be safe let's set the loading state back to false before going home
```suggestion
      this.loading = false;
      this.$router.push('/');
```